src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -174,6 +174,10 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
 
                     return NullHostedService.Instance;
                 });
+
+                // Wire this up early so that any early worker logs are guaranteed to be flushed if any other
+                // IHostedService has a slow startup.
+                services.AddSingleton<IHostedService, WorkerConsoleLogService>();
```

Comment:
This is a second issue with debugging. The log coming from the worker identifying the target process id was being written, but never flushed, if we were waiting for a debugger to attach *and* the host wasn't able to start up fully because of that. 

This happens, for example, when RunOnStartup is enabled for timer triggers.